config/routes.rb

```diff
@@ -7,6 +7,17 @@
     resources :sessions, only: [:new, :destroy]
   end
 
+  resources :users, only: [] do
+    scope module: :hackathon do
+      resources :subscriptions, only: :index do
+        get "unsubscribe_all", on: :collection
+      end
+      namespace :subscriptions do
+        resource :bulk, controller: :bulk, only: [:update, :destroy]
+      end
```

Comment:
Instead of having a custom action and a separate controller for all subscriptions, how about just acting on all of the user's subscriptions if "id" is `all`? So `DELETE /subscriptions/x` or `DELETE /subscriptions/all`